src/test/java/net/snowflake/client/jdbc/telemetryOOB/TelemetryServiceIT.java

```diff
@@ -231,4 +240,76 @@ public void stressTestCreateUrgentEvent()
     }
     sw.stop();
   }
+
+  private void generateDummyException(int value, SFSession session) throws SnowflakeSQLLoggedException
+  {
+    if (value > 0)
+    {
+      String queryID = "01234567-1234-1234-1234-00001abcdefg";
+      String reason = "This is a test exception.";
+      String sqlState = SqlState.NO_DATA;
+      int vendorCode = value;
+      throw new SnowflakeSQLLoggedException(queryID, reason, sqlState, vendorCode, session);
+    }
+  }
+  
+  /**
+   * Test case for checking telemetry message for SnowflakeSQLExceptions. Assert that telemetry OOB endpoint is reached
+   * after a SnowflakeSQLLoggedException is thrown.
+   * @throws SQLException
+   */
+  @Test
+  public void testSnowflakeSQLLoggedExceptionOOBTelemetry() throws SQLException, InterruptedException
+  {
+    // make a connection to initialize telemetry instance
+    Connection con = getConnection();
+    int fakeErrorCode = 27;
+    try
+    {
+      generateDummyException(fakeErrorCode, null);
+    }
+    catch (SnowflakeSQLLoggedException e)
+    {
+      // The error response has the same code as the the fakeErrorCode
+      assertThat("Communication error", e.getErrorCode(),
+              equalTo(fakeErrorCode));
+
+      // since it returns normal response,
+      // the telemetry does not create new event
+      Thread.sleep(WAIT_FOR_TELEMETRY_REPORT_IN_MILLISECS);
+      if (TelemetryService.getInstance().isDeploymentEnabled())
+      {
+        assertThat("Telemetry event has not been reported successfully. Error: " +
+                        TelemetryService.getInstance().getLastClientError(),
+                TelemetryService.getInstance().getClientFailureCount(), equalTo(0));
+      }
+      return;
+    }
+    fail();
+  }
+
+  /**
+   * Test case for checking telemetry message for SnowflakeSQLExceptions. In-band telemetry should be used.
+   *
+   * @throws SQLException
+   */
+  @Test
+  public void testSnowflakeSQLLoggedExceptionIBTelemetry() throws SQLException
+  {
+    // make a connection to initialize telemetry instance
+    Connection con = getConnection();
+    int fakeErrorCode = 27;
+    try
+    {
+      generateDummyException(fakeErrorCode, con.unwrap(SnowflakeConnectionV1.class).getSfSession());
+    }
+    catch (SnowflakeSQLLoggedException e)
+    {
+      // The error response has the same code as the fakeErrorCode
+      assertThat("Communication error", e.getErrorCode(),
+              equalTo(fakeErrorCode));
+      return;
+    }
+    fail();
```

Comment:
you can put `fail();` under `generateDummyException` and then you don't need `return;` in `catch`.

Reply:
Changed!